icepyx/core/read.py

```diff
@@ -372,14 +438,49 @@ def vars(self):
 
         if not hasattr(self, "_read_vars"):
             self._read_vars = Variables(
-                "file", path=self._filelist[0], product=self._prod
+                "file", path=self.filelist[0], product=self.product
             )
 
         return self._read_vars
+    
+    @property
+    def filelist(self):
+        """
+        Return the list of files represented by this Read object.
+        """
+        return self._filelist
+    
+    @property
+    def num_files(self):
+        """
+        Return the number of files that are being processed.
+        """
+        return len(self.filelist)
```

Comment:
Is there a longer term motivation for having this property available? Wondering if we can trim out a function and just show the user `len(reader.filelist)`.

Reply:
nope, no foresight here. Just something I was doing a lot and considered. I'll remove it.